src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDateFormat = date.split(fromFormat[fromFormat.length - 1]);
+  const newDateFormat = [];
+
+  for (let i = 0; i < oldDateFormat.length; i++) {
+    if (toFormat.indexOf(fromFormat[i]) === -1) {
+      if (fromFormat[i].length === 2) {
+        if (+oldDateFormat[i] < 30) {
+          newDateFormat[toFormat.indexOf('YYYY')] = '20' + oldDateFormat[i];
+        } else {
+          newDateFormat[toFormat.indexOf('YYYY')] = '19' + oldDateFormat[i];
+        }
+      } else {
+        newDateFormat[toFormat.indexOf('YY')] = oldDateFormat[i].slice(-2);
```

Comment:
The condition to check the length of the format element to determine if it's a year is not robust. It would be more reliable to directly compare the string value of the format element (e.g., 'YYYY' or 'YY') to decide on the conversion strategy.